integration/obscurogateway/obscurogateway_test.go

```diff
@@ -197,3 +467,165 @@ func waitServerIsReady(serverAddr string) error {
+
+	client := &http.Client{}
+	response, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+	return io.ReadAll(response.Body)
+}
+
+func prepareRegisterPayload(userID string, pk *ecdsa.PrivateKey, hexAddress string) string {
+	message := fmt.Sprintf("Register %s for %s", userID, strings.ToLower(hexAddress))
+	prefixedMessage := fmt.Sprintf("\u0019Ethereum Signed Message:\n%d%s", len(message), message)
+	messageHash := crypto.Keccak256([]byte(prefixedMessage))
+	sig, err := crypto.Sign(messageHash, pk)
+	if err != nil {
+		fmt.Printf("Failed to sign message: %v\n", err)
+	}
+	sig[64] += 27
+	signature := "0x" + hex.EncodeToString(sig)
+	payload := fmt.Sprintf("{\"signature\": \"%s\", \"message\": \"%s\"}", signature, message)
+	return payload
+}
+
+func ComputeContractAddress(sender gethcommon.Address, nonce uint64) (gethcommon.Address, error) {
+	// RLP encode the byte array of the sender's address and nonce
+	encoded, err := rlp.EncodeToBytes([]interface{}{sender, nonce})
+	if err != nil {
+		return gethcommon.Address{}, err
+	}
+	// Compute the Keccak-256 hash of the RLP encoded byte array
+	hash := crypto.Keccak256(encoded)
+
+	// The contract address is the last 20 bytes of this hash
+	return gethcommon.BytesToAddress(hash[len(hash)-20:]), nil
+}
+
+func TransferETHToAddress(client *ethclient.Client, wallet wallet.Wallet, toAddress gethcommon.Address, amount int64) error {
+	transferTx1 := types.LegacyTx{
+		Nonce:    wallet.GetNonceAndIncrement(),
+		To:       &toAddress,
+		Value:    big.NewInt(amount),
+		Gas:      uint64(1_000_000),
+		GasPrice: gethcommon.Big1,
+		Data:     nil,
+	}
+	signedTx, err := wallet.SignTransaction(&transferTx1)
+	if err != nil {
+		return err
+	}
+	err = client.SendTransaction(context.Background(), signedTx)
+	return err
+}
+
+func InteractWithSmartContract(client *ethclient.Client, wallet wallet.Wallet, contractAbi abi.ABI, methodName string, methodParam string, contractAddress gethcommon.Address) (gethcommon.Hash, error) {
+	contractInteractionData, err := contractAbi.Pack(methodName, methodParam)
+	if err != nil {
+		return gethcommon.Hash{}, err
+	}
+
+	interactionTx := types.LegacyTx{
+		Nonce:    wallet.GetNonceAndIncrement(),
+		To:       &contractAddress,
+		Gas:      uint64(1_000_000),
+		GasPrice: gethcommon.Big1,
+		Data:     contractInteractionData,
+	}
+	signedTx, err := wallet.SignTransaction(&interactionTx)
+	if err != nil {
+		return gethcommon.Hash{}, err
+	}
+	err = client.SendTransaction(context.Background(), signedTx)
+	if err != nil {
+		return gethcommon.Hash{}, err
+	}
+
+	return signedTx.Hash(), nil
+}
+
+type User struct {
+	UserID            string
+	Wallets           []wallet.Wallet
+	HTTPClient        *ethclient.Client
+	WSClient          *ethclient.Client
+	ServerAddressHTTP string
+	ServerAddressWS   string
+}
+
+func NewUser(wallets []wallet.Wallet, serverAddressHTTP string, serverAddressWS string) (*User, error) {
```

Comment:
This seems to be a perfect start for a `tools/walletextention/lib` a library that allows to programatically access the resources of the gateway.

Perhaps it's not important to do this now, maybe splitting this `User` struct into its own file would be a good change for now ?

Reply:
I completely agree that it would be nice to have `tools/walletextention/lib`.
Will split it in separate file and create a task in to create library in the future